apps/meteor/ee/server/apps/communication/rest.ts

```diff
@@ -1124,14 +1124,42 @@ export class AppsRestApi {
 						return API.v1.failure('Invalid status provided, it must be "status" field and a string.');
 					}
 
-					const prl = manager.getOneById(this.urlParams.id);
+					const baseUrl = orchestrator.getMarketplaceUrl();
+					const { id: appId } = this.urlParams;
+					const { version, status } = this.bodyParams;
+
+					const headers = getDefaultHeaders();
+
+					const mktAppsUrl = new URL(`${baseUrl}/v1/apps/${appId}`);
+
+					mktAppsUrl.searchParams.set('appVersion', String(version));
+
+					const request = await fetch(mktAppsUrl.toString(), {
+						headers,
+					});
+
+					if (!request.ok) {
+						return API.v1.failure('Unable to retrieve app from marketplace');
+					}
```

Comment:
Can you confirm what `request.ok` means? The marketplace-api will respond with a `404` if the appId is not found. This could prevent private apps from having their status changed via the user